openstudiocore/src/model/FileOperations.cpp

```diff
@@ -79,242 +79,69 @@
-
-    // ensure directory exists
-    QFileInfo dstInfo(dstPath);
-    QString fileName = dstInfo.fileName();
-    if (!dstInfo.exists() || !dstInfo.isDir()){
-      QDir parentDstDir(QFileInfo(dstPath).path());
-      LOG_FREE(Info, "copyDir", "Creating directory named = '" << toString(fileName) << "' in parentDstDir = '" << toString(parentDstDir.path()) << "'");
-      if (!parentDstDir.mkpath(fileName)){
-        LOG_FREE(Error, "copyDir", "Failed to create directory = '" << toString(fileName) << "' in parentDstDir = '" << toString(parentDstDir.path()) << "'");
-        return false;
-      }
+    if (!openstudio::filesystem::create_directory(destinationDir))
+    {
+      throw std::runtime_error("Cannot create destination directory " + destinationDir.string());
     }
 
-    QDir srcDir(srcPath);
 
-    // remove all files in dst as well as any directories in dst that are not in src
-    test = synchDirStructures(srcPath, dstPath);
-    if (!test){
-      LOG_FREE(Error, "copyDir", "Failed to synch destination '" << toString(dstPath) << "' with source '" << toString(srcPath) << "'");
-      result = false;
-    }
 
-    // copy all files in src to dst
-    for (const QFileInfo &srcItemInfo : srcDir.entryInfoList(QDir::Dirs | QDir::Files | QDir::NoDotAndDotDot))
+    for (const auto& dirEnt : openstudio::filesystem::recursive_directory_iterator{sourceDir})
     {
-      QString srcItemPath = srcPath + "/" + srcItemInfo.fileName();
-      QString dstItemPath = dstPath + "/" + srcItemInfo.fileName();
-
-      QFileInfo dstItemInfo = QFileInfo(dstItemPath);
-
-      if (srcItemInfo.isDir()) {
-        test = copyDir(srcItemPath, dstItemPath);
-        if (!test){
-          LOG_FREE(Error, "copyDir", "Failed to copy directory '" << toString(srcItemPath) << "' to '" << toString(dstItemPath) << "'");
-          result = false;
-
-          // DLM: do we really want to give up here?
-          //return false;
-        }
-      } else if (srcItemInfo.isFile()) {
-        test = QFile::copy(srcItemPath, dstItemPath);
-        if (!test){
-          LOG_FREE(Error, "copyDir", "Failed to copy file '" << toString(srcItemPath) << "' to '" << toString(dstItemPath) << "'");
-          result = false;
-
-          // DLM: do we really want to give up here?
-          //return false;
-        }
+      const auto& path = dirEnt.path();
+      auto relativePathStr = path.string();
+      boost::replace_first(relativePathStr, sourceDir.string(), "");
+
+      try {
+        openstudio::filesystem::copy(path, destinationDir / relativePathStr);
+      } catch (const std::exception &e) {
+        LOG_FREE(Error, "copyDir", "Error copying from: " << toString(path) << " to: " << toString(destinationDir / relativePathStr) << " Description: " << e.what());
+        return false;
       }
     }
 
-    return result;
   }
 
-#ifdef Q_OS_WIN
-
-#include <windows.h>
-
-QString longPathName(const QString& path)
```

Comment:
Do we not need this longPathName anymore?

Reply:
Not with the code as it stands, if we go back to the older more complex synch behavior this will get re-added.